### PR TITLE
ACS chat: Api changes jan 2021

### PR DESCRIPTION
**Pull request by danielgerlag**
- Pulled in the latest swagger from the INT Api
- Ran AutoRest against the new swagger
- Updated breaking changes
- Added test for HTML content
- Recorded new test sessions for playback

### CODE REVIEW

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/ChatMessage.java [hunk 1]
```diff
@@ -21,14 +21,9 @@
...
      *
-     * Possible values:
-     * - Text
-     * - ThreadActivity/TopicUpdate
-     * - ThreadActivity/AddMember
-     * - ThreadActivity/DeleteMember
      */
     @JsonProperty(value = "type")
-    private String type;
+    private ChatMessageType type;
```
Comment (angiurgiu) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/ChatMessage.java:26]: Did you forget about "required = true", or did you leave it out on purpose?

Comment (angiurgiu) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/ChatMessage.java:26]: Same for other properties

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/ChatMessageType.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
```
Comment (angiurgiu) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/ChatMessageType.java:3]: nit: it's not generated code

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/SendChatMessageOptions.java [hunk 3]
```diff
@@ -29,6 +29,12 @@
     @JsonProperty(value = "senderDisplayName")
     private String senderDisplayName;
 
+    /*
```
Comment (angiurgiu) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/SendChatMessageOptions.java:32]: Let's specify here that the only types of messages that can be sent are 'text' and 'html'

File: sdk/communication/azure-communication-chat/src/samples/java/com/azure/communication/chat/ReadmeSamples.java [hunk 4]
```diff
@@ -85,7 +85,7 @@ public void createChatThread() {
             .setTopic("Topic")
             .setParticipants(participants);
         CreateChatThreadResult result = chatClient.createChatThread(createChatThreadOptions);
-        String chatThreadId = result.getThread().getId();
+        String chatThreadId = result.getChatThread().getId();
```
Comment (angiurgiu) [inline sdk/communication/azure-communication-chat/src/samples/java/com/azure/communication/chat/ReadmeSamples.java:88]: I think this also requires a README update

**Review by angiurgiu — CHANGES_REQUESTED** [review]

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/ChatMessage.java (hunk 1 above)
Comment (danielgerlag) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/ChatMessage.java:26]: I didn't know about it, is there a release note for the api that covers this?

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/SendChatMessageOptions.java (hunk 3 above)
Comment (danielgerlag) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/SendChatMessageOptions.java:32]: Do you mean on the public setter or this private field?

Comment (angiurgiu) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/SendChatMessageOptions.java:32]: On the setter

**Review by angiurgiu — APPROVED** [review]
LGM

**Pull request merged**